src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java

```diff
@@ -139,8 +140,115 @@ public void api_get_id_returns_a_subject_that_does_not_exist() throws Exception
+        verify(ucsbsubjectRepository, times(1)).save(correctUCSBSubject);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals(expectedReturn, responseString);
+    }
+
+    // Test api /put with Subject id that doesn't exist
+    @WithMockUser(roles = { "ADMIN" })
+    @Test
+    public void api_subject_put_that_does_not_exist() throws Exception {
+        // arrange
+
+        UCSBSubject updatedUCSBSubject = UCSBSubject.builder()
+                .subjectTranslation("Test Subject Translation")
+                .deptCode("Test Department Code")
+                .collegeCode("Test College Code")
+                .subjectCode("Test Subject Code")
+                .relatedDeptCode("Test related department code")
+                .inactive(true)
+                .id(77L)
+                .build();
+
+        String requestBody = mapper.writeValueAsString(updatedUCSBSubject);
+
+        when(ucsbsubjectRepository.findById(eq(77L))).thenReturn(Optional.empty());
+
+        // act
+        MvcResult response = mockMvc.perform(
+                put("/api/UCSBSubjects?id=77")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .characterEncoding("utf-8")
+                        .content(requestBody)
+                        .with(csrf()))
+                .andExpect(status().isBadRequest()).andReturn();
+
+        // assert
+        verify(ucsbsubjectRepository, times(1)).findById(77L);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals("ucsb subject with id 77 not found", responseString);
+    }
+
+        //test delete endpoint
+        @WithMockUser(roles = { "ADMIN" })
+        @Test
+        public void api_subject__delete_subject() throws Exception {
+                // arrange
+
+                UCSBSubject subject1 = UCSBSubject.builder()
+                .subjectTranslation("Test Subject Translation")
+                .deptCode("Test Department Code")
+                .collegeCode("Test College Code")
+                .subjectCode("Test Subject Code")
+                .relatedDeptCode("Test related department code")
+                .inactive(true)
+                .id(15L)
+                .build();
+
+                when(ucsbsubjectRepository.findById(eq(15L))).thenReturn(Optional.of(subject1));
+
+                // act
+                MvcResult response = mockMvc.perform(
+                        delete("/api/UCSBSubjects?id=15")
+                                .with(csrf()))
+                        .andExpect(status().isOk()).andReturn();
+
+                // assert
+                verify(ucsbsubjectRepository, times(1)).findById(15L);
+                verify(ucsbsubjectRepository, times(1)).deleteById(15L);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals("ucsb subject with id 15 deleted", responseString);
+        }
+
+
```

Comment:
Oh, I think you are missing the test case where the user wants to delete a subject that does not exist